agent/provider/src/cli/rule.rs

```diff
@@ -85,19 +88,29 @@ fn set(set_rule: SetRule, config: ProviderConfig) -> Result<()> {
             SetOutboundRule::Partner(RuleWithCert::CertId { cert_id, mode }) => {
                 rules.set_partner_mode(cert_id, mode)
             }
-            SetOutboundRule::Partner(RuleWithCert::ImportCert { import_cert, mode }) => {
+            SetOutboundRule::Partner(RuleWithCert::ImportCert {
+                import_cert,
+                mode,
+                whole_chain,
+            }) => {
                 let mut keystore = CompositeKeystore::load(&rules.cert_dir)?;
 
-                let AddResponse { added, skipped } = keystore.add(&AddParams {
+                let AddResponse {
+                    added,
+                    skipped,
+                    leaf_cert_ids,
+                } = keystore.add(&AddParams {
                     certs: vec![import_cert],
                     permissions: vec![CertPermissions::All],
-                    whole_chain: true,
+                    whole_chain,
                 })?;
 
                 rules.keystore.reload(&rules.cert_dir)?;
-
+                let leaf_cert_ids: HashSet<String> = HashSet::from_iter(leaf_cert_ids.into_iter());
                 for cert in added.into_iter().chain(skipped) {
-                    rules.set_partner_mode(cert.id(), mode.clone())?;
+                    if whole_chain || leaf_cert_ids.contains(&cert.id()) {
+                        rules.set_partner_mode(cert.id(), mode.clone())?;
+                    }
```

Comment:
Or even change whole approach:
```rust
if whole_chain {
    for cert in added {
        <set>
    }
} else {
    <set for first ID as it would be sorted from leaf to root>
}
```

Reply:
Did not wanted to have two places where partner rule is being set, but changed as requested.
Have chosen first option, because you want to apply rule to every `leaf_cert_ids` (every Golem cert is a leaf node).